homeassistant/components/zigbee.py

```diff
@@ -0,0 +1,283 @@
+    def bool2state(self):
+        """
+        A dictionary mapping booleans to GPIOSetting objects to translate
+        on/off as being pin high or low.
+        """
+        return self._bool2state
+
+    @property
+    def state2bool(self):
+        """
+        A dictionary mapping GPIOSetting objects to booleans to translate
+        pin high/low as being on or off.
+        """
+        return self._state2bool
+
+# Create an alias so that ZigBeeDigitalOutConfig has a logical opposite.
+ZigBeeDigitalInConfig = ZigBeeDigitalPinConfig
+
+
+class ZigBeeDigitalOutConfig(ZigBeeDigitalPinConfig):
+    """
+    A subclass of ZigBeeDigitalPinConfig which sets _should_poll to default as
+    False instead of True. The value will still be overridden by the presence
+    of a 'poll' config entry.
+    """
+    def __init__(self, config):
+        super(ZigBeeDigitalOutConfig, self).__init__(config)
+        self._should_poll = config.get("poll", False)
+
+
+class ZigBeeAnalogInConfig(ZigBeePinConfig):
+    """
+    Handles the fetching of configuration from the config file for a ZigBee
+    GPIO pin set to analog in.
+    """
+    @property
+    def max_voltage(self):
+        """
+        The voltage at which the ADC will report its highest value.
+        """
+        return float(self._config.get("max_volts", DEFAULT_ADC_MAX_VOLTS))
+
+
+class ZigBeeDigitalIn(ToggleEntity):
+    """
+    ToggleEntity to represent a GPIO pin configured as a digital input.
+    """
+    def __init__(self, hass, config):
+        self._config = config
+        self._state = False
+        if config.should_poll:
+            hass.pool.add_job(JobPriority.EVENT_STATE, (self.update, None))
+
+    @property
+    def name(self):
+        return self._config.name
+
+    @property
+    def should_poll(self):
+        return self._config.should_poll
+
+    @property
+    def is_on(self):
+        return self._state
+
+    def update(self, *args):
+        """
+        Ask the ZigBee device what its output is set to.
+        """
+        pin_state = DEVICE.get_gpio_pin(
+            self._config.pin,
+            self._config.address)
+        self._state = self._config.state2bool[pin_state]
+        self.update_ha_state()
+
+
+class ZigBeeDigitalOut(ZigBeeDigitalIn):
+    """
+    Adds functionality to ZigBeeDigitalIn to control an output.
+    """
+    def __init__(self, hass, config):
+        super(ZigBeeDigitalOut, self).__init__(hass, config)
+        # Get initial value regardless of whether we should_poll.
+        # If config.should_poll is True, then it's already been handled in
+        # our parent class __init__().
+        if not config.should_poll:
+            hass.pool.add_job(JobPriority.EVENT_STATE, (self.update, None))
+
+    def _set_state(self, state):
+        DEVICE.set_gpio_pin(
+            self._config.pin,
+            self._config.bool2state[state],
+            self._config.address)
+        self._state = state
+        self.update_ha_state()
```

Comment:
Yes, I wasn't sure about this. You wouldn't want to have polling switched on for outputs if HA is the only thing controlling them. However, you would probably still want an initial poll to check their current state when HA starts.

It almost requires a separate config item which would poll for initial state while leaving regular polling turned off. I doubt this is unique to ZigBee though, so is there a HA-wide way of solving this problem?